### PR TITLE
Make create folder popup support nested folders

### LINKED ISSUE #6678: Create Folder popup should support nested folders
### Describe the project you are working on

This applies to pretty much all projects no matter how small.

### Describe the problem or limitation you are having in your project

The *Create Folder* popup in the *FileSystem* window doesn't support creating nested folders. 

Often I'm wanting to organise my assets into various subfolders such as characters/player, props/door etc. For larger projects this nesting can get much deeper.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow use of the / character in the *Create Folder* popup to create nested folders.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- If I type *foo/bar* and *foo* folder doesn't exist it will be created followed by the *bar* folder inside of it.
- If I type *foo/bar* and the *foo* folder already exists but *bar* doesn't, bar will be created inside *foo*.
- If I type *foo/bar* and the *foo/bar* folder already exists, we get the standard 'File with that name already exists.' error we get now.

Exact same functionality with regards to capitalisation, duplicates etc that we have now.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

There are two ways to work arount hsi currently:
- Create these nested folders one by one in Godot (and wait for the scanning process to complete each time). This is very slow.
- Do it outside of Godot in file explorer.

Neither option is optimal.

### Is there a reason why this should be core and not an add-on in the asset library?

It feels natural for the feature to exist within the editor. The fact that it's missing feels more like a papercut than anything. I'm not sure the existing functionality can be modified with an add-on but I'm not familiar enough with the codebase to know for sure.

### PR DESCRIPTION
**Pull request by timothyqiu**
Closes https://github.com/godotengine/godot-proposals/issues/6678

Peek 2023-04-16 11-21

### CODE REVIEW

**Comment by adamscott** [comment]
I'm not sure about the new lines in the description. 

I know that showing `existing/foo/bar/another/` would be the same as the input, but it feels weird for me and it takes much space.

**Comment by KoBeWi** [comment]
It's still weird xd
The info could just say "subfolders will be created recursively" or something.

**Comment by timothyqiu** [comment]
Updated. The hierarchical preview is removed.

File: editor/directory_create_dialog.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	const String error = _validate_path(path);
+	ERR_FAIL_COND_MSG(!error.is_empty(), error);
+
+	Error err;
+	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+
+	err = da->change_dir(base_dir);
+	ERR_FAIL_COND_MSG(err != OK, "Cannot open directory '" + base_dir + "'.");
+
+	print_verbose("Making folder " + path + " in " + base_dir);
```
Comment (KoBeWi) [inline editor/directory_create_dialog.cpp:109]:
```suggestion
	print_verbose("Creating folder " + path + " in " + base_dir);
```

File: editor/directory_create_dialog.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+void DirectoryCreateDialog::_post_popup() {
+	dir_path->grab_focus();
+}
+
+void DirectoryCreateDialog::config(const String &p_base_dir) {
+	base_dir = p_base_dir;
+	label->set_text(vformat(TTR("Create new folder in %s:"), base_dir));
+	dir_path->set_text("new folder");
```
Comment (KoBeWi) [inline editor/directory_create_dialog.cpp:127]: Shouldn't this be translated?

File: editor/directory_create_dialog.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	_on_dir_path_changed(dir_path->get_text());
+}
+
+void DirectoryCreateDialog::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("dir_created"));
+}
+
+void DirectoryCreateDialog::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
```
Comment (KoBeWi) [inline editor/directory_create_dialog.cpp:138]:
```suggestion
```
`NOTIFICATION_THEME_CHANGED` is received automatically on tree enter.

**Comment by KoBeWi** [comment]
The default height can be smaller:
image

File: editor/directory_create_dialog.cpp (hunk 2 above)
Comment (timothyqiu) [reply editor/directory_create_dialog.cpp:127]: The name was not translated in the old dialog.

I'm not sure if it is fixed now, but Android export had trouble loading non-ascii paths / filenames.

**Comment by timothyqiu** [comment]
> The default height can be smaller

Updated code. `Window::popup_centered()` now correctly decides that the height should be 159px. But the window adjusted to around 200px _after_ popup. Any idea what could be the cause?

**Comment by KoBeWi** [comment]
Maybe try using `reset_size()` before popup?

**Comment by timothyqiu** [comment]
The strange initial height is caused by `Label` having an unexpected min height when set to `AUTOWRAP_WORD`.

Setting the status label to not autowrap for now. I'll open an issue for the behavior :(

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**